src/box/replication.cc

```diff
@@ -1342,6 +1313,7 @@ replicaset_sync(void)
 		say_info("replica set sync complete");
 		box_set_orphan(false);
 	}
+	fiber_testcancel();
```

Comment:
Should be checked after `fiber_cond_wait_deadline`? Anyway, do we need to raise an exception in this case while we could just return as we do in case we failed to form a quorum?